src/transformers/generation_utils.py

```diff
@@ -689,6 +691,8 @@ def generate(
         num_beam_groups: Optional[int] = None,
         diversity_penalty: Optional[float] = None,
         prefix_allowed_tokens_fn: Optional[Callable[[int, torch.Tensor], List[int]]] = None,
+        logits_processor: Optional[LogitsProcessorList] = None,
```

Comment:
Hmm, not super happy about this - is it really needed to define `logits_processor` outside of `generate()` ? What would be the use case for this? We would then also have to add `logits_warper` to it. Also what do we do then if a user say passes `no_repeat_ngram_size` and passes a `logits_processor` with `no_repeat_ngram_size`? Which one should we take? 

=> I would prefer to not add them to the arguments of `generate()`. What do you think?

Reply:
I've had a discussion where users where actually using custom LogitsProcessor (like penalize certain tokens and whatnot, a bit like bad_words, but seemed even more specialized). 

It made sense, but this could very well go in another PR (if we even desire it)

Reply:
And the behavior is all logits processor will run, it's the caller's responsability to know what is in it IMO. Like stopping_criteria.

Reply:
I removed it from this PR.